sdk/tables/azure-data-tables/inc/azure/data/tables/table_client.hpp

```diff
@@ -128,21 +128,29 @@ namespace Azure { namespace Data { namespace Tables {
      */
     Response<Models::DeleteEntityResult> DeleteEntity(
         Models::TableEntity const& tableEntity,
-        Core::Context const& context = {});
+        Core::Context const& context = {}) const;
 
     /**
-     * @brief Upsert specified entity in a table.
+     * @brief Update or insert specified entity in a table.
      *
-     * @param tableEntity The TableEntity to upsert.
-     * @param upsertKind Upsert kind(update/merge), default update.
+     * @param tableEntity The TableEntity to update or insert.
      * @param context for canceling long running operations.
      * @return Upsert entity result.
      */
-    Response<Models::UpsertEntityResult> UpsertEntity(
+    Response<Models::UpdateEntityResult> UpdateInsertEntity(
```

Comment:
Do we expect there to be no difference between the returned value of `UpdateInsertEntity()` vs `UpdateEntity()` (and same question with Merge)?

We are returning `UpdateEntityResult` in both cases. @gearama Is there a chance the service could diverge in what is being returned in the future, and if so, should we future proof it to be different return types?

If the returned model/headers will always be identical, then aside from a naming wart, it seems reasonable to keep the models identical. Though, outside of resources generated from response bodies, we typically don't re-use `OperationResult` return types across multiple "operations".

@LarryOsterman what do you think?

Reply:
highly doubt it , when that happens we'll figure it out

Reply:
Okay.
Let's keep in mind we can't overload in C++ based on the return type, if it were to diverge. Hopefully the chance is low :)